packages/sw-precaching/src/lib/controllers/base-cache-manager.js

```diff
@@ -14,6 +14,7 @@ class BaseCacheManager {
   constructor(cacheName) {
     this._entriesToCache = new Map();
     this._cacheName = cacheName;
+    this._requestWrapper = new RequestWrapper({cacheName: this._cacheName});
```

Comment:
Rather than always constructing your own `RequestWrapper`, how about allowing developers to optionally pass in their own?

Developers might want to do that to allow them to assign a `BroadcastCacheUpdate` behavior to their `RequestWrapper`, so that they could announce to controlled pages when there's an update to a precached resource. It could also be useful to override the `CacheableResponse` behavior if they have custom criteria for determining whether a precaching response is valid.

One problem I see is figuring out what would happen if the pass in both a `cacheName` and a `RequestWrapper` that already has a differently named cache.